### PR TITLE
refactor(tree): Enable compressed encoding in pending tests

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Enables compressed encoding for some tests that were missing a schema.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

The task calls for

> Essentially we want a way to use our validator for our tests (like how our customers would)

But I think that's not really what the tests [AB#7111](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7111) was about needed, since they're operating at a level where the schema validator doesn't apply (they're creating a `SharedTree` object and directly using its `editor` (non-exposed API) to create changes, not schematizing it and using public APIs of `ITree`.

### CODE REVIEW

File: packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts [hunk 1]
```diff
@@ -400,11 +419,17 @@ describe("SharedTree benchmarks", () => {
...
-							for (let iCommit = 0; iCommit < commitCount; iCommit++) {
-								for (let iPeer = 0; iPeer < peerCount; iPeer++) {
-									const peer = provider.trees[iPeer];
+							for (let iPeer = 0; iPeer < peerCount; iPeer++) {
+								const peer = provider.trees[iPeer];
+								updateSchema(peer, schemaGeneralized);
+								// Can't use assertSchema() because it sets an event listener that fails when we're rebasing
+								// the changes that move from one schema to another, because we try to go from an optional rootField
+								// to a forbidden one.
+								// assertSchema(peer, schemaGeneralized);
```
Comment (alexvy86) [inline packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:428]: Can we make it so the schema change is outside of the range of changes/commits that get rebased, so we can call `assertSchema()` like the other tests?

File: packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts [hunk 2]
```diff
@@ -341,6 +345,12 @@ describe("SharedTree benchmarks", () => {
 
 	describe("acking local commits", () => {
 		const localCommitSize = [1, 25, 100, 500, 1000];
+		const builder = new SchemaBuilderBase(FieldKinds.optional, {
+			scope: "test",
+			libraries: [leaf.library],
+		});
+		const schemaGeneralized = builder.intoSchema(Any);
```
Comment (WayneFerrao) [inline packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:352]: So before this change, was there no schema available for sharedtree(  including a generalized one)? Was this generalized schema  (as opposed to flexfield/rootField) recently introduced?

Comment (alexvy86) [reply packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:352]: To be honest I'm not entirely clear how it works :) just followed an existing pattern in other tests that used compressed encoding. I'm hoping the SharedTree team will point out if there's anything weird with the usage pattern.

File: packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts [hunk 3]
```diff
@@ -1886,11 +1888,17 @@ describe("SharedTree", () => {
...
+export function assertSchema<TRoot extends FlexFieldSchema>(
 	tree: SharedTree,
 	schema: FlexTreeSchema<TRoot>,
 	onDispose: () => void = () => assert.fail(),
 ): FlexTreeView<TRoot> {
 	const viewSchema = new ViewSchema(defaultSchemaPolicy, {}, schema);
 	return requireSchema(tree.checkout, viewSchema, onDispose, createMockNodeKeyManager());
 }
+
+export function updateSchema(tree: SharedTree, schema: FlexTreeSchema): void {
```
Comment (Josmithr) [inline packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts:1900]: Docs please :)

File: packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts [hunk 4]
```diff
@@ -66,7 +70,7 @@ const nodesCountDeep = [
 // TODO: ADO#7111 Schema should be fixed to enable schema based encoding.
```
Comment (seanimam) [inline packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:70]: Can this comment be removed now?

File: packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts [hunk 5]
```diff
@@ -66,7 +70,7 @@ const nodesCountDeep = [
...
 const factory = new SharedTreeFactory({
 	jsonValidator: typeboxValidator,
-	treeEncodeType: TreeCompressionStrategy.Uncompressed,
+	treeEncodeType: TreeCompressionStrategy.Compressed,
```
Comment (seanimam) [inline packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:73]: Is there a need to keep benchmarks for a tree without compressed encoding? Wondering if we should maintain tests for both.

File: packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts [hunk 6]
```diff
@@ -355,6 +365,10 @@ describe("SharedTree benchmarks", () => {
 						const provider = new TestTreeProviderLite(1, factory);
 						// TODO: specify a schema for these trees.
 						const [tree] = provider.trees;
+
+						updateSchema(tree, schemaGeneralized);
+						assertSchema(tree, schemaGeneralized);
```
Comment (seanimam) [inline packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts:370]: I am not familiar with how these tests work exactly but I'm wondering if schema updates and assertions should occur after the inserts on line 373?

**Pull request closed without merging**